agentendpoint/config_task.go

```diff
@@ -328,22 +342,38 @@ func (c *configTask) run(ctx context.Context) error {
 				break
 			}
 			if enforcementActionTaken := enforceConfigResourceState(ctx, plcy, rCompliance, configResource); enforcementActionTaken {
-				// On any change we trigger post check.
-				c.postCheckRequired = true
+				// On any change we trigger post check for all previous resouces.
+				c.markPostCheckRequired()
 			}
 			if plcy.hasError {
 				break
 			}
 		}
 	}
 
-	if c.postCheckRequired {
-		c.postCheckState(ctx)
-	}
+	// Run any post checks that we need to.
+	c.postCheckState(ctx)
 
 	return c.reportCompletedState(ctx, "", agentendpointpb.ApplyConfigTaskOutput_SUCCEEDED)
 }
 
+// Mark all resources that have already completed as "needs post check".
+func (c *configTask) markPostCheckRequired() {
+	for _, osPolicy := range c.Task.GetOsPolicies() {
+		plcy, ok := c.policies[osPolicy.GetId()]
+		if !ok || plcy.hasError {
```

Comment:
Consider adding a comment about when the `!ok` condition will be met. It might not be immediately obvious when reading this in the future.

Also, a comment about why we ignore policies with errors will also help the reasons (if and when we want to modify this logic) in the future.